src/sdk/namespace_fs.js

```diff
@@ -1560,6 +1602,89 @@ class NamespaceFS {
         //     throw new RpcError('INVALID_OBJECT_STATE', 'Attempted to access archived or sparse file');
         // }
     }
+    //////////////////////////
+    //// VERSIONING UTILS ////
+    //////////////////////////
+
+    _is_versioning_enabled() {
+        return this.versioning === VER_ENABLED;
+    }
+
+    _is_versioning_disabled() {
+        return this.versioning === VER_DISABLED;
+    }
+
+    _get_version_id_by_stat({ino, mtimeBigint}) {
+        // TODO: GPFS might require generation number to be added to version_id
+        return 'mtime-' + mtimeBigint.toString(36) + '-ino-' + ino.toString(36);
+    }
+
+    // 1. if version_id_str is null version - nothing to etract
+    // 2. else extract the attributes from it or fail for invalid version_id_str
+    // version_id_str - mtime-{mtimeBigint}-ino-{ino} | explicit null
+    // returns mtimeBigint, ino (inode_number)
+    _extract_version_attr_from_xattr(version_id_str) {
+        if (version_id_str === 'null') return;
+        const arr = version_id_str.split('mtime-').join('').split('-ino-');
+        if (arr.length < 2) throw new Error('Invalid version_id_string, cannot extact attributes');
+        return { mtimeBigint: this.string_to_bigint(arr[0], 36), ino: parseInt(arr[1], 36) };
+    }
+
+    _get_version_id_by_xattr(stat) {
+       return (stat && stat.xattr[XATTR_VERSION_ID]) || 'null';
+    }
+
+    _get_version_path(key, version_id) {
+        const key_version = path.basename(key) + (version_id ? '_' + version_id : '');
+        return path.normalize(path.join(this.bucket_path, path.dirname(key), hidden_versions_path, key_version));
+    }
+
+    string_to_bigint(str, base) {
+        let res = 0n;
+        const arr = str.split('');
+        for (const char of arr) {
+            res = BigInt(parseInt(char, base)) + (BigInt(base) * res);
+        }
+        return res;
+    }
+
+    // this function returns the following version attributes -
+    // version_id_str - mtime-{mtimeBigint}-ino-{ino} | explicit null
+    // mtimeBigint - modified timestmap in bigint - last time the content of the file was modified
+    // ino - refers to the data stored in a particular location
+    // if xattr contains version attr - return attr of xattr
+    // else - it's a null version - return stat
+    async _get_version_attr(fs_context, version_path) {
```

Comment:
let's change function name to _get_version_info. attr may make us confused with xattr.